prow/plugins/golint/suggestion/golint_suggestion.go

```diff
@@ -150,6 +152,18 @@ func fixRanges(p lint.Problem) string {
 	return suggestion
 }
 
+func fixVarDecl(p lint.Problem) string {
```

Comment:
Do we want to create a meta-func for translating a `lint.Problem` into a suggestion?

Reply:
Yeah, I was thinking that bit and maybe doing the match and passing in just the matched bits? Up to you

Reply:
Sounds good. It does make sense to have the regex matching part in `SuggestCodeChange`

Reply:
related comments: https://github.com/kubernetes/test-infra/pull/10050#discussion_r232841878 https://github.com/kubernetes/test-infra/pull/10050#discussion_r233643865

While less DRY, it might be nicer for unit testing to include the regexp matching inside the suggestion handler instead of just passing the handler the matches? Either option is fine with me though. This repetition is minimal and doesn't significantly impact complexity or cognitive load.

Reply:
@cjwagner I see. Let's move the regex matching into `SuggestCodeChange` for now